src/utils/embedded/embedded.utils.ts

```diff
@@ -205,6 +205,50 @@ async function insecurelyValidateApplication() {
 // Validate immediately on load
 // insecurelyValidateApplication();
 
+const getAbsoluteUrl = (url: string): string | null => {
+  if (!url) {
+    return null;
+  }
+  try {
+    return new URL(url, window.location.origin).href;
+  } catch {
+    return null;
+  }
+};
+
+export function getAppLogo(): string | null {
+  try {
```

Comment:
You could consider getting the icon from the manifest before trying to get it from the page. See https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Manifest